Xwt/Xwt/WindowFrame.cs

```diff
@@ -83,7 +84,7 @@ protected override void OnBackendCreated ()
 				base.OnBackendCreated ();
 				Parent.location = Backend.Bounds.Location;
 				Parent.size = Backend.Bounds.Size;
-				Backend.EnableEvent (WindowFrameEvent.BoundsChanged);
+				OnEnableEvent (WindowFrameEvent.BoundsChanged);
```

Comment:
what if we do this only if `owned == true`? Enabling this by default is only required by the initial window pisition/size logic afair (@slluis? correct me if I'm wrong). If the client needs all the bounds/sizing feature, it should enable/disable this event manually.

Reply:
We also use that event to keep the Bounds property up to date, so I guess we still need it.

Reply:
Hhm, right, but in that case we could query `Backend.Bounds` directly. And since we don't need `Reallocate ()` if the window is embedded this should do the trick I think:

```diff
--- a/Xwt/Xwt/WindowFrame.cs
+++ b/Xwt/Xwt/WindowFrame.cs
@@ -84,7 +84,8 @@ namespace Xwt
 				base.OnBackendCreated ();
 				Parent.location = Backend.Bounds.Location;
 				Parent.size = Backend.Bounds.Size;
-				OnEnableEvent (WindowFrameEvent.BoundsChanged);
+				if (Parent.owned)
+					OnEnableEvent (WindowFrameEvent.BoundsChanged);
 			}
 			
 			public void OnBoundsChanged (Rectangle bounds)
@@ -404,7 +405,9 @@ namespace Xwt
 		internal virtual Rectangle BackendBounds {
 			get {
 				BackendHost.EnsureBackendLoaded ();
-				return new Rectangle (location, size);
+				if (owned)
+					return new Rectangle (location, size);
+				return Backend.Bounds;
 			}
 			set {
 				size = value.Size;

```